src/twitch-service.ts

```diff
@@ -59,120 +80,111 @@ export class TwitchEventSubClient extends Context.Tag("twitch-eventsub-client")<
-        await Effect.logInfo("Ban event").pipe(Effect.runPromise);
-      });
-
-      eventsub.onChannelChatMessage(
-        config.broadcasterId,
-        config.broadcasterId,
-        async (event) => {
-          await Effect.logInfo("chat message").pipe(Effect.runPromise);
-
-          await Effect.gen(function* () {
-            if (event.messageText.startsWith("!")) {
-              if (event.messageText === "!song") {
-                yield* pubsub.publish(Message.CurrentlyPlayingRequest());
-              }
-            }
-          }).pipe(Effect.runPromise);
-        },
-      );
-
-      // eventsub.onChannelRedemptionAddForReward(
-      //   config.broadcasterId,
-      //   config.songRequestRewardId,
-      //   (event) => {
-      //     Effect.gen(function* () {
-      //       yield* pubsub.publish(Message.SongRequest({ uri: event.input }));
-      //     });
-      //   },
-      // );
-
-      yield* Effect.forkScoped(
-        Effect.forever(
-          Effect.gen(function* () {
-            yield* Effect.logInfo("waiting for song");
-            const { song } = yield* Queue.take(dequeue);
-            yield* Effect.logInfo("received song");
-
-            yield* Effect.tryPromise(async () => {
-              return api.chat.sendChatMessage(
-                config.broadcasterId,
-                `The currently playing song is ${song.name}`,
-              );
-            });
-          }),
-        ),
+export const TwitchService = Layer.scopedDiscard(
+  Effect.gen(function* () {
+    yield* Effect.logInfo("twitch service starting");
+    const api = yield* Twitch;
+    const config = yield* TwitchConfig;
+    const eventsub = yield* TwitchEventSubClient;
+    const pubsub = yield* MessagePubSub;
+
+    const dequeue = yield* pubsub.subscribeTo("CurrentlyPlaying");
+
+    eventsub.onChannelBan(config.broadcasterId, async () => {
+      await Effect.logInfo("Ban event").pipe(Effect.runPromise);
+    });
+
+    eventsub.onChannelChatMessage(
+      config.broadcasterId,
+      config.broadcasterId,
+      async (event) => {
+        await Effect.logInfo("chat message").pipe(Effect.runPromise);
+
+        if (event.messageText === "!song") {
+          await Effect.runPromise(
+            pubsub.publish(Message.CurrentlyPlayingRequest())
+          );
+        }
+      }
+    );
+
+    // eventsub.onChannelRedemptionAddForReward(
+    //   config.broadcasterId,
+    //   config.songRequestRewardId,
+    //   (event) => {
+    //     Effect.gen(function* () {
+    //       yield* pubsub.publish(Message.SongRequest({ uri: event.input }));
+    //     });
+    //   },
+    // );
+
+    yield* Effect.gen(function* () {
+      yield* Effect.logInfo("waiting for song");
+      const { song } = yield* Queue.take(dequeue);
+      yield* Effect.logInfo("received song");
+      yield* api.use((_) =>
+        _.chat.sendChatMessage(
+          config.broadcasterId,
+          `The currently playing song is ${song.name}`
+        )
```

Comment:
I was wondering where `use` was going to show up, this is a great little addition